CMakeLists.txt

```diff
@@ -232,6 +232,26 @@ if(WIN32)
     endif(CMAKE_SIZEOF_VOID_P EQUAL 8)
 endif(WIN32)
 
+# Setup FMA Capabilities Flag
+IF(UNIX)
+    MESSAGE(STATUS "Checking for FMA support...")
+    IF(NOT APPLE)
+        EXEC_PROGRAM(cat ARGS "/proc/cpuinfo" OUTPUT_VARIABLE CPUINFO)
+    ELSE(NOT APPLE)
+        EXEC_PROGRAM("/usr/sbin/sysctl -n machdep.cpu.features" OUTPUT_VARIABLE CPUINFO)
+    ENDIF(NOT APPLE)
+
+    STRING(REGEX REPLACE "^.*(fma).*$" "fma" FMA_THERE ${CPUINFO})
+    STRING(COMPARE EQUAL "fma" "${FMA_THERE}" FMA_TRUE)
+    IF (FMA_TRUE)
+		MESSAGE(STATUS "Found FMA support.  Setting fma compile flag.")
+		set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -mfma")
```

Comment:
Don’t do this! This allows the compiler to issue FMA instructions. The resulting code will not run on a machine without support for FMA. Think of a Linux distribution using a fancy build machine and some user installing the packing on a machine without FMA.

Reply:
rear1019, the if process there is a detection system to determine if FMA is available on the system or not, so if it doesn't support it, it won't be set.  I have an Udoo board for instance that doesn't support FMA so the modules built on there don't issue FMA instructions and everything runs fine.  Do you have an instance where it could false-positive detect FMA that's not covered in the detection logic?

Reply:
It detects on the build system, not the system where the binaries get executed (what if the Debian build server supports FMA, then all Debians would need it).